tsdoc/src/parser/StringChecks.ts

```diff
@@ -2,12 +2,36 @@
  * Helpers for validating various text string formats.
  */
 export class StringChecks {
-  private static readonly tsdocTagNameRegExp: RegExp = /^@[a-z][a-z0-9]*$/i;
+  private static readonly _tsdocTagNameRegExp: RegExp = /^@[a-z][a-z0-9]*$/i;
 
-  private static readonly urlSchemeRegExp: RegExp = /^[a-z][a-z0-9+\-.]*\:/i;
+  private static readonly _urlSchemeRegExp: RegExp = /^[a-z][a-z0-9+\-.]*\:/i;
+
+  private static readonly _identifierNotWordCharRegExp: RegExp = /\W/u;
+  private static readonly _identifierNumberStartRegExp: RegExp = /^[0-9]/u;
+
+  private static readonly _reservedWords: Set<string> = new Set<string>([
```

Comment:
>_reservedWords [](start = 26, length = 14)

+ "module", "async", "of", "exports (maybe)"

Reply:
Modern JavaScript interpreters generally allow most of these names, even as the name of a function.  I realized that the main motivation for this rule was a concern about names such as `MyClass.(static:instance)` where it's unclear which side of the colon has the selector.

Thus I have relaxed the validation to look for TSDoc system selectors rather than general ECMAScript reserved words.

---
In reply to: [221410121](https://github.com/Microsoft/tsdoc/pull/75#discussion_r221410121) [](ancestors = 221410121)